index.ts

```diff
@@ -1,17 +1,96 @@
 import type { AstroIntegration } from "astro";
-import fs from "fs/promises";
+import { cpSync, rmSync } from "fs";
+import { join } from "path";
 
 const PKG_NAME = "astro-preload";
 
-export default function preload(): AstroIntegration {
+/**
+ * A rather basic vite plugin that exposes config data needed by
+ * the preload-Function.
+ * @param config Config Data from the Astro config
+ */
+function makeVitePlugin(config: {publicDir: string, base: string}) {
+
+    return {
+        name: "vite-plugin-"+PKG_NAME,
+        resolveId(id: string) : string | void {
+            if(id === "virtual:astro-preload") {
+                return `\0${id}`
+            }
+        },
+        load(id: string) : string | void {
+            if(id === "\0virtual:astro-preload") {
+                const moduleContent = Object.entries(config).map( ([k,v]) => `export const ${k} = ${JSON.stringify(v)}`).join("\n")
+                console.log(moduleContent)
+                return moduleContent
+            }
+        }
+    }
+}
+
+let pluginConfig: {
+    publicDir: string,
+    base: string,
+    site?: string | undefined,
+    outDir: string
+} | undefined
+
+interface Config {
+    reuseGenerated?: boolean | undefined
+}
+
+
+export default function makePlugin(config?: Config): AstroIntegration {
     return {
         name: PKG_NAME,
         hooks: {
-            "astro:build:done": async () => {
-                await fs.mkdir("dist/assets/preloaded", { recursive: true });
-                const files = await fs.readdir("public/assets/preloaded");
-                await Promise.all(files.map(async file => fs.copyFile(`public/assets/preloaded/${file}`, `dist/assets/preloaded/${file}`)));
-            }
+            // Taken from starlight (compare with https://github.com/withastro/starlight/blob/main/packages/starlight/index.ts#L42-L57)
+            "astro:config:setup": ({config, updateConfig}) => {
+                const vitePlugins = config?.vite?.plugins ?? []
+                const pubDir = config.publicDir
+                console.log(pubDir.protocol)
+                pluginConfig = {
+                    publicDir: config.publicDir.toString().replace(pubDir.protocol+"//", ""),
```

Comment:
`url.replace(pubDir.protocol+"//", "")` is reaaally hacky, but I am not sure how to do it better. If I just `toString` the URL, cp'ing files breaks because it assumes that `file://` is not a protocol, but a (rather malformed) folder called `file:`

Same with L56. 
If you have any suggestions, let me know